compiler/next/lib/frontend/Parser/chpl.ypp

```diff
@@ -646,18 +638,30 @@ toplevel_stmt:
 pragma_ls:
   TPRAGMA STRINGLITERAL
     {
-      SizedStr ss = $2;
-      $$ = new UniqueStrList();
-      PODUniqueString u = STR(ss.allocatedData);
-      delete ss.allocatedData;
-      $$->push_back(u);
+      Expression* lit = $2;
+      if (lit->isStringLiteral()) {
```

Comment:
It seems off to do a check on the result of an all caps item to me.  It makes me worry we're losing information in the else branch case

Reply:
Specifically, it looks like we're losing error message information?

Reply:
I'm not sure what you mean? Suppose there is an error in a string literal and ErroneousExpression is returned by `$2`. Then we ignore the pragma it would have made and the error is still recorded (in the list of parser errors).

Reply:
Ah, okay.  That seems a little unclear from the structure, maybe a comment in the else branch would help?

Reply:
Sure, I can add a comment.